components/gantry/oneaxis/oneaxis.go

```diff
@@ -294,89 +139,32 @@ func (g *oneAxis) homeEncoder(ctx context.Context) error {
 	positionB := positionA + revPerLength
 
 	g.positionLimits = []float64{positionA, positionB}
-	return nil
-}
-
-func (g *oneAxis) rotationalToLinear(positions float64) float64 {
-	theRange := g.positionLimits[1] - g.positionLimits[0]
-	x := positions / g.lengthMm
-	x = g.positionLimits[0] + (x * theRange)
-	g.logger.Debugf("oneAxis SetPosition %.2f -> %.2f", positions, x)
-
-	return x
-}
 
-func (g *oneAxis) testLimit(ctx context.Context, zero bool) (float64, error) {
-	defer utils.UncheckedErrorFunc(func() error {
-		return g.motor.Stop(ctx, nil)
-	})
-
-	d := -1.0
-	if !zero {
-		d *= -1
+	if g.gantryRange = positionB - positionA; g.gantryRange == 0 {
+		return errZeroLengthGantry
 	}
 
-	err := g.motor.GoFor(ctx, d*g.rpm, 0, nil)
-	if err != nil {
-		return 0, err
-	}
-
-	start := time.Now()
-	for {
-		hit, err := g.limitHit(ctx, zero)
-		if err != nil {
-			return 0, err
-		}
-		if hit {
-			err = g.motor.Stop(ctx, nil)
-			if err != nil {
-				return 0, err
-			}
-			break
-		}
-
-		elapsed := start.Sub(start)
-		if elapsed > (time.Second * 15) {
-			return 0, errors.New("gantry timed out testing limit")
-		}
-
-		if !utils.SelectContextOrWait(ctx, time.Millisecond*10) {
-			return 0, ctx.Err()
-		}
-	}
-
-	return g.motor.Position(ctx, nil)
-}
-
-func (g *oneAxis) limitHit(ctx context.Context, zero bool) (bool, error) {
-	offset := 0
-	if !zero {
-		offset = 1
-	}
-	pin, err := g.board.GPIOPinByName(g.limitSwitchPins[offset])
-	if err != nil {
-		return false, err
-	}
-	high, err := pin.Get(ctx, nil)
-
-	return high == g.limitHigh, err
+	return nil
 }
 
 // Position returns the position in millimeters.
 func (g *oneAxis) Position(ctx context.Context, extra map[string]interface{}) ([]float64, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
 	pos, err := g.motor.Position(ctx, extra)
 	if err != nil {
 		return []float64{}, err
 	}
 
-	theRange := g.positionLimits[1] - g.positionLimits[0]
-	x := g.lengthMm * ((pos - g.positionLimits[0]) / theRange)
-
+	// never divide by zero since we cannot create a gantry with a zero range
+	x := g.lengthMm * ((pos - g.positionLimits[0]) / g.gantryRange)
 	return []float64{x}, nil
 }
 
 // Lengths returns the physical lengths of an axis of a Gantry.
 func (g *oneAxis) Lengths(ctx context.Context, extra map[string]interface{}) ([]float64, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
```

Comment:
I doubt locking the mutex will hurt anything, but how could it help anything, either? I'd sure expect that everything used in this function could be considered immutable, in which case there are no race conditions to avoid.